python-checks/src/test/java/org/sonar/python/checks/PandasDataFrameToNumpyCheckTest.java

```diff
@@ -20,13 +20,81 @@
 package org.sonar.python.checks;
 
 import org.junit.jupiter.api.Test;
+import org.sonar.python.checks.quickfix.PythonQuickFixVerifier;
 import org.sonar.python.checks.utils.PythonCheckVerifier;
 
 class PandasDataFrameToNumpyCheckTest {
 
+  PandasDataFrameToNumpyCheck check = new PandasDataFrameToNumpyCheck();
+
+  private static final String QUICK_FIX_MESSAGE = "Replace this with a call to DataFrame.to_numpy()";
+
   @Test
   void test() {
-    PythonCheckVerifier.verify("src/test/resources/checks/pandasDataFrameToNumpy.py", new PandasDataFrameToNumpyCheck());
+    PythonCheckVerifier.verify("src/test/resources/checks/pandasDataFrameToNumpy.py", check);
+  }
+
+  @Test
+  void quick_fix_test_1() {
+    final String non_compliant = "def non_compliant_1(xx):\n" +
+      "    import pandas as pd\n" +
+      "\n" +
+      "    df = pd.DataFrame({\n" +
+      "        'X': ['A', 'B', 'A', 'C'],\n" +
+      "        'Y': [10, 7, 12, 5]\n" +
+      "    })\n" +
+      "\n" +
+      "    _ = df.values";
+    final String compliant = "def non_compliant_1(xx):\n" +
+      "    import pandas as pd\n" +
+      "\n" +
+      "    df = pd.DataFrame({\n" +
+      "        'X': ['A', 'B', 'A', 'C'],\n" +
+      "        'Y': [10, 7, 12, 5]\n" +
+      "    })\n" +
+      "\n" +
+      "    _ = df.to_numpy()";
+    performVerification(non_compliant, compliant);
+  }
+
+  @Test
+  void quick_fix_test_2() {
+
+    final String non_compliant = "def non_compliant_1(xx):\n" +
+      "    import pandas as pd\n" +
+      "\n" +
+      "    _ = pd.DataFrame({\n" +
+      "        'X': ['A', 'B', 'A', 'C'],\n" +
+      "        'Y': [10, 7, 12, 5]\n" +
+      "    }).values";
+    final String compliant = "def non_compliant_1(xx):\n" +
+      "    import pandas as pd\n" +
+      "\n" +
+      "    _ = pd.DataFrame({\n" +
+      "        'X': ['A', 'B', 'A', 'C'],\n" +
+      "        'Y': [10, 7, 12, 5]\n" +
+      "    }).to_numpy()";
+    performVerification(non_compliant, compliant);
   }
 
+  @Test
+  void quick_fix_test_3() {
+
+    final String non_compliant = "def dataframe_from_read_csv():\n" +
+      "    import pandas as pd\n" +
+      "\n" +
+      "    my_df = pd.read_csv(\"some_csv.csv\")\n" +
+      "    my_df.values";
+    final String compliant = "def dataframe_from_read_csv():\n" +
+      "    import pandas as pd\n" +
+      "\n" +
+      "    my_df = pd.read_csv(\"some_csv.csv\")\n" +
+      "    my_df.to_numpy()";
+    performVerification(non_compliant, compliant);
+  }
+
+  private void performVerification(String non_compliant, String compliant) {
+    PythonQuickFixVerifier.verify(check, non_compliant, compliant);
+    PythonQuickFixVerifier.verifyQuickFixMessages(check, non_compliant, QUICK_FIX_MESSAGE);
```

Comment:
This is a nitpick but I generally prefer to have the quick fix message tested explicitly (i.e having it written as a string literal rather than using the field from the check). This makes things more intentional and protects us against typos.

I know we don't consistently do it and I don't think it matters too much so I'm not asking for a change here, just sharing an opinion on how to approach these tests.